sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1642,9 +1642,10 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
       Set(Map("a" -> "10", "b" -> "p"), Map("a" -> "20", "b" -> "c"), Map("a" -> "3", "b" -> "p")))
 
     // table to alter does not exist
-    intercept[NoSuchTableException] {
+    val e = intercept[AnalysisException] {
```

Comment:
Why did you replace `NoSuchTableException `?

Reply:
Unresolved table is not handled here:
https://github.com/apache/spark/blob/8e2633962f789a6ba5eb9448596f6ac4b7b1c2ff/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L104-L105

Reply:
then we should throw `NoSuchTableException` here.

Reply:
otherwise with such approach, you will lift all specific exception to the general one - `AnalysisException`.

Reply:
let's fix it separately.

Reply:
Sounds good. I will do a follow-up a PR.